doc/lua_api.md

```diff
@@ -7351,11 +7355,17 @@ child will follow movement and rotation of that bone.
     * If `continuous` is true, the Lua entity will not be moved to the current
       position before starting the interpolated move.
     * For players this does the same as `set_pos`,`continuous` is ignored.
-* `punch(puncher, time_from_last_punch, tool_capabilities, direction)`
-    * `puncher` = another `ObjectRef`,
-    * `time_from_last_punch` = time since last punch action of the puncher
-    * `direction`: can be `nil`
-* `right_click(clicker)`; `clicker` is another `ObjectRef`
+    * no-op if object is attached
+* `punch(puncher, time_from_last_punch, tool_capabilities, dir)`
+    * punches the object, triggering all consequences a normal punch would have
+    * `puncher`: another `ObjectRef` which punched the object
+    * `dir`: direction vector of punch (can be `nil`)
+    * Other arguments: See `on_punch` for entities
```

Comment:
Please add that the args (apart from puncher) get default values if `nil`. Otherwise the reader will think that `nil` is propagated to `on_punch`.